src/pages/Wallet/AddressesPage.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import addressToGroup from 'alephium-js/dist/lib/address'
+import { TOTAL_NUMBER_OF_GROUPS } from 'alephium-js/dist/lib/constants'
+import { deriveNewAddressData } from 'alephium-js/dist/lib/wallet'
+import { AnimatePresence } from 'framer-motion'
+import { useEffect, useState } from 'react'
+import styled from 'styled-components'
+
+import ActionLink from '../../components/ActionLink'
+import Amount from '../../components/Amount'
+import Button from '../../components/Button'
+import Label from '../../components/Label'
+import Modal from '../../components/Modal'
+import { ContainerNextToSidebar } from '../../components/PageComponents/PageContainers'
+import PageTitle from '../../components/PageComponents/PageTitle'
+import Table, { AlignType, TableCell, TableFooter, TableRow } from '../../components/Table'
+import { useGlobalContext } from '../../contexts/global'
+import NewAddressPage from './NewAddressPage'
+
+const minTableColumnWidth = '105px'
+
+type AddressesTableData = {
+  hash: string
+  isMain: boolean
+  label?: {
+    text?: string
+    color?: string
+  }
+  lastUsed: string
+  transactions: number
+  tokens: number
+  group: number
+  amount: bigint
+}
+
+const AddressesPage = () => {
+  const [isGenerateNewAddressModalOpen, setIsGenerateNewAddressModalOpen] = useState(false)
+  const { addressesInfo, wallet, client } = useGlobalContext()
+  const [addressesTableData, setAddressesTableData] = useState<AddressesTableData[]>([])
+  const headers = [
```

Comment:
This is fully static and hence could be defined outside of the component.
I wonder if there's a performance impact? If not, I'm also fine with packing it inside there. What is your opinion?

Also, we could define the type to avoid mistakes:  `headers: TableProps['headers'] = ...`

Reply:
Good catch. I suppose that every time the component rerenders, a new location in memory is being used to define `const headers`. I would guess that the garbage collector would clean the other one, but in any case I agree with you that it's better to move it out of the component.

Reply:
Updated.